framework/db/ActiveRelationTrait.php

```diff
@@ -278,7 +279,14 @@ public function populateRelation($name, &$primaryModels)
             $buckets = $this->indexBuckets($buckets, $this->indexBy);
         }
 
-        $link = array_values(isset($viaQuery) ? $viaQuery->link : $this->link);
+        $link = array_values($this->link);
+        if (isset($viaQuery)) {
+            $deepViaQuery = $viaQuery;
+            while ($deepViaQuery->via) {
+                $deepViaQuery = is_array($deepViaQuery->via) ? $deepViaQuery->via[1] : $deepViaQuery->via;
```

Comment:
~~~php
is_array($deepViaQuery->via)
~~~
doesn't garantee that there is element with index 1

Reply:
Unnecessary checks. Format of `ActiveRelationTrait::via` property is fixed `[$relationName, $relation]` and must have index 1 if its type is Array. I can throw exception, but need add that check everywhere.